pkg/networkservice/common/discovercrossnse/metadata.go

```diff
@@ -14,17 +14,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package interpose
+package discovercrossnse
 
-import "strings"
+import (
+	"context"
 
-const nameSuffix = "#interpose-nse"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+)
 
-func interposeName(name string) string {
-	return name + nameSuffix
+type selectedForworderKey struct{}
+
+func loadForwarderName(ctx context.Context) string {
+	v, ok := metadata.Map(ctx, false).Load(selectedForworderKey{})
+	if !ok {
+		return ""
+	}
+	return v.(string)
 }
 
-// Is returns true if passed name contains interpose identity
-func Is(name string) bool {
-	return strings.HasSuffix(name, nameSuffix)
+func storeForwarderName(ctx context.Context, v string) {
```

Comment:
@edwarnicke What if we'll load forwarder name from the path segments instead of metadata?

Reply:
How would that work on initial request?

Reply:
>How would that work on initial request?

For the initial request, we do have not the next segment. 
So simply go to discover a forwarder (see at branch where `forwarderName == ""`)

Reply:
And what if the next segment is no longer valid?

Reply:
@edwarnicke  That's a good question. It looks like in this case we should select another forwawrder.